core/node/node_framework/src/implementations/layers/logs_bloom_backfill.rs

```diff
@@ -0,0 +1,61 @@
+use zksync_logs_bloom_backfill::LogsBloomBackfill;
+
+use crate::{
+    implementations::resources::pools::{MasterPool, PoolResource},
+    service::StopReceiver,
+    task::{Task, TaskId, TaskKind},
+    wiring_layer::{WiringError, WiringLayer},
+    FromContext, IntoContext,
+};
+
+/// Wiring layer for ethereum watcher
+///
+/// Responsible for initializing and running of [`LogsBloomBackfill`] task, that backfills `logsBloom` for old blocks.
+#[derive(Debug)]
+pub struct LogsBloomBackfillLayer;
+
+#[derive(Debug, FromContext)]
+#[context(crate = crate)]
+pub struct Input {
+    pub master_pool: PoolResource<MasterPool>,
+}
+
+#[derive(Debug, IntoContext)]
+#[context(crate = crate)]
+pub struct Output {
+    #[context(task)]
+    pub logs_bloom_backfill: LogsBloomBackfill,
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for LogsBloomBackfillLayer {
+    type Input = Input;
+    type Output = Output;
+
+    fn layer_name(&self) -> &'static str {
+        "logs_bloom_backfill_layer"
+    }
+
+    async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
+        let main_pool = input.master_pool.get_singleton().await.unwrap();
```

Comment:
Makes sense to propagate an error here instead of `unwrap`ping.

Reply:
copied from another place, now propagated in both